packages/flutter/lib/src/material/theme_data.dart

```diff
@@ -560,7 +563,9 @@ class ThemeData with Diagnosticable {
     // COMPONENT THEMES
     appBarTheme ??= const AppBarTheme();
     bannerTheme ??= const MaterialBannerThemeData();
-    bottomAppBarTheme ??= const BottomAppBarTheme();
+    bottomAppBarTheme ??= BottomAppBarTheme(
```

Comment:
To make this change non-breaking, we'd want to keep this as was before and rely on Dart fix instead

Reply:
Done - thanks for the discussion!